app/services/submission_creator.rb

```diff
@@ -37,4 +37,11 @@ def save(submission, answers)
       a.save
     end
   end
+
+  def submission_and_answers_are_valid
+    # so that it runs :valid? on all of the objects and sets errors if necessary
+    submission_is_valid = submission.valid?
+    answers_are_valid = answers.map(&:valid?).all?
```

Comment:
Yeah, its because '#all?' is lazy - as soon as it finds an element that fails a predicate it returns false. And we need to go through all of them. ;)